src/main/java/com/couchbase/intellij/tools/dialog/ImportDialogController.java

```diff
@@ -0,0 +1,571 @@
+package com.couchbase.intellij.tools.dialog;
+
+import static utils.ProcessUtils.printOutput;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.swing.event.DocumentEvent;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.couchbase.client.java.json.JsonArray;
+import com.couchbase.client.java.json.JsonObject;
+import com.couchbase.intellij.database.ActiveCluster;
+import com.couchbase.intellij.tools.CBTools;
+import com.couchbase.intellij.workbench.Log;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.ui.DocumentAdapter;
+
+import utils.FileUtils;
+
+public class ImportDialogController {
+
+    protected static final String[] possibleScopeFields = { "cbms", "scope", "cbs" };
+    protected static final String[] possibleCollectionFields = { "cbmc", "collection", "cbc" };
+    protected static final String[] possibleKeyFields = { "cbmk", "cbmid", "key", "cbk" };
+
+    protected static String targetScopeField;
+    protected static String targetCollectionField;
+
+    protected static void addListeners(ImportDialog importDialog) {
```

Comment:
Updatesummary should called whenever a entity is updated. It should scan all values and show the result.